providers/azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/options/AlertRequestOptions.java

```diff
@@ -0,0 +1,268 @@
+
+		return buildQueryParameters().equals(options.buildQueryParameters());
+	}
+
+	@Override
+	public int hashCode() {
+		return buildQueryParameters().hashCode();
+	}
+
+	public static class Builder {
+
+		public static AlertRequestOptions targetResource(String targetResource) {
+			return new AlertRequestOptions().withTargetResource(targetResource);
+		}
+
+		public static AlertRequestOptions targetResourceGroup(String targetResourceGroup) {
+			return new AlertRequestOptions().withTargetResourceGroup(targetResourceGroup);
+		}
+
+		public static AlertRequestOptions targetResourceGroupType(String targetResourceGroupType) {
+			return new AlertRequestOptions().withTargetResourceType(targetResourceGroupType);
+		}
+
+		public static AlertRequestOptions monitorService(String monitorService) {
+			return new AlertRequestOptions().withMonitorService(monitorService);
+		}
+
+		public static AlertRequestOptions monitorCondition(String monitorCondition) {
+			return new AlertRequestOptions().withMonitorCondition(monitorCondition);
+		}
+
+		public static AlertRequestOptions severity(String severity) {
+			return new AlertRequestOptions().withSeverity(severity);
+		}
+
+		public static AlertRequestOptions alertState(String alertState) {
+			return new AlertRequestOptions().withAlertState(alertState);
+		}
+
+		public static AlertRequestOptions alertRule(String alerRule) {
+			return new AlertRequestOptions().withAlertRule(alerRule);
+		}
+
+		public static AlertRequestOptions smartGroupId(String smartGroupId) {
+			return new AlertRequestOptions().withSmartGroupId(smartGroupId);
+		}
+
+		public static AlertRequestOptions includeContext(Boolean includeContext) {
+			return new AlertRequestOptions().withIncludeContext(includeContext);
+		}
+
+		public static AlertRequestOptions includeEgressConfig(Boolean includeEgressConfig) {
+			return new AlertRequestOptions().withIncludeEgressConfig(includeEgressConfig);
+		}
+
+		public static AlertRequestOptions pageCount(Integer pageCount) {
+			return new AlertRequestOptions().withPageCount(pageCount);
+		}
+
+		public static AlertRequestOptions sortBy(String sortBy) {
+			return new AlertRequestOptions().withSortBy(sortBy);
+		}
+
+		public static AlertRequestOptions sortOrder(String sortOrder) {
+			return new AlertRequestOptions().withSortOrder(sortOrder);
+		}
+
+		public static AlertRequestOptions select(String select) {
+			return new AlertRequestOptions().withSelect(select);
+		}
+
+		public static AlertRequestOptions timeRange(String timeRange) {
+			return new AlertRequestOptions().withTimeRange(timeRange);
+		}
+
+		public static AlertRequestOptions customTimeRange(String customTimeRange) {
+			return new AlertRequestOptions().withCustomTimeRange(customTimeRange);
+		}
+
+		public static AlertRequestOptions groupBy(String groupBy) {
+			return new AlertRequestOptions().withGroupBy(groupBy);
+		}
+
+		public static AlertRequestOptions includeSmartGroupsCount(Boolean includeSmartGroupsCount) {
```

Comment:
Please convert as many of these to primitives as possible.

Reply:
Updated as per comment